src/pybind/mgr/volumes/module.py

```diff
@@ -0,0 +1,376 @@
+            'perm': 'r'
+        },
+        {
+            'cmd': 'fs volume rm '
+                   'name=vol_name,type=CephString',
+            'desc': "Delete a CephFS volume",
+            'perm': 'r'
+        },
+        {
+            'cmd': 'fs subvolume create '
+                   'name=vol_name,type=CephString '
+                   'name=sub_name,type=CephString '
+                   'name=size,type=CephString,req=false ',
+            'desc': "Create a CephFS subvolume within an existing volume",
+            'perm': 'r'
+        },
+        {
+            'cmd': 'fs subvolume rm '
+                   'name=vol_name,type=CephString '
+                   'name=sub_name,type=CephString',
+            'desc': "Delete a CephFS subvolume",
+            'perm': 'r'
+        },
+
+        # volume ls [recursive]
+        # subvolume ls <volume>
+        # volume authorize/deauthorize
+        # subvolume authorize/deauthorize
+
+        # volume describe (free space, etc)
+        # volume auth list (vc.get_authorized_ids)
+
+        # snapshots?
+
+        # FIXME: we're doing CephFSVolumeClient.recover on every
+        # path where we instantiate and connect a client.  Perhaps
+        # keep clients alive longer, or just pass a "don't recover"
+        # flag in if it's the >1st time we connected a particular
+        # volume in the lifetime of this module instance.
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        self._initialized = Event()
+
+        self._background_jobs = Queue.Queue()
+
+    def _oremote(self, *args, **kwargs):
+        """
+        Helper for invoking `remote` on whichever orchestrator is enabled
+        """
+        # TODO: orchestrator_cli has a setting for orchestrator backend,
+        # but we don't really want to start propagating that into
+        # other modules individually -- maybe mgrmap should have
+        # the setting?  or just a C++-level mgr config option?
+
+        return self.remote("rook",
+                           *args, **kwargs)
+
+    def serve(self):
+        # TODO: discover any subvolumes pending purge, and enqueue
+        # them in background_jobs at startup
+
+        # TODO: consume background_jobs
+        #   skip purge jobs if their fscid no longer exists
+
+        # TODO: on volume delete, cancel out any background jobs that
+        # affect subvolumes within that volume.
+
+        # ... any background init needed?  Can get rid of this
+        # and _initialized if not
+        self._initialized.set()
+
+    def handle_command(self, inbuf, cmd):
+        self._initialized.wait()
+
+        handler_name = "_cmd_" + cmd['prefix'].replace(" ", "_")
+        try:
+            handler = getattr(self, handler_name)
+        except AttributeError:
+            return -errno.EINVAL, "", "Unknown command"
+
+        return handler(inbuf, cmd)
+
+    def _orchestrator_wait(self, completions):
```

Comment:
This seems to be an outdated copy of `orchestrator_cli.module.OrchestratorCli#_wait`. Maybe, we should unify this into a single `OrchestratorClient` class.

Reply:
Sounds good to me... can that be a follow-on cleanup?